gulpfile.js

```diff
@@ -99,32 +121,38 @@ var generatedTargetDir = path.join(
 var generatedCssTargetDir = path.join(generatedTargetDir, 'css');
 var generatedJsTargetDir = path.join(generatedTargetDir, 'js');
 
-for (var dependencyId in frontendDependencies) {
-  var dependency = frontendDependencies[dependencyId];
-  var dependencyDir = dependency.targetDirPrefix + dependency.version;
-  if (dependency.hasOwnProperty('bundle')) {
-    if (dependency.bundle.hasOwnProperty('css')) {
-      dependency.bundle.css.forEach(function(cssFiles) {
-        cssFilesPath.push(path.join(
-          'third_party', 'static', dependencyDir, cssFiles));
-      });
-    }
-    if (dependency.bundle.hasOwnProperty('js')) {
-      dependency.bundle.js.forEach(function(jsFiles) {
-        jsFilesPath.push(path.join(
-          'third_party', 'static', dependencyDir, jsFiles));
-      });
-    }
-    if (dependency.bundle.hasOwnProperty('fontsPath')) {
-      var fontPrefix = '*.{eot,woff2,ttf,woff,eof,svg}';
-      fontFolderPath.push(path.join('third_party', 'static', dependencyDir,
-        dependency.bundle.fontsPath, fontPrefix));
-    }
-    if (dependency.bundle.hasOwnProperty('cssBackgroundImage')) {
-      dependency.bundle.cssBackgroundImage.forEach(function(imagePath) {
-        cssBackgroundPath.push(path.join(
-          'third_party', 'static', dependencyDir, imagePath));
-      });
+for(var downloadFormat in frontendDependencies) {
```

Comment:
I'm not sure I understand why this is top-level -- why don't we do the validity checks only at the point of generating the dependencies? Would it be better for this to only be in a function that only runs when a generation task is executed (which is really when you need the manifest to be valid)?